### PR TITLE
Add transifex config

### PR DESCRIPTION
**Pull request by MorrisJobke**
Fixes https://github.com/brantje/nextnote/issues/16

### CODE REVIEW

File: .tx/config [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[main]
+host = https://www.transifex.com
+lang_map = bg_BG: bg, cs_CZ: cs, fi_FI: fi, hu_HU: hu, nb_NO: nb, sk_SK: sk, th_TH: th, ja_JP: ja
```
Comment (enoch85) [inline .tx/config:3]: I don't know, but should't all languages be there? And if not, why?

Comment (MorrisJobke) [reply .tx/config:3]: This is only the mapping of special ones to the general ones, where we don't have the general ones. So this is fine and used in all apps.

**Comment by MorrisJobke** [comment]
@nextcloud-bot needs to be admin to be able to push to master.

**Review by brantje — APPROVED** [review]

**Review by enoch85 — APPROVED** [review]

**Comment by enoch85** [comment]
> @nextcloud-bot needs to be admin to be able to push to master.

@brantje Do you fix that?

**Comment by brantje** [comment]
Ya already done =)

**Comment by MorrisJobke** [comment]
Ah ... pushing the existing stuff isn't that easy sadly :/

**Comment by brantje** [comment]
You have my permission to nuke it. Translations aren't yet build in into the frontend.

**Comment by MorrisJobke** [comment]
Ah ... okay, then we will start with a fresh one from transifex.

**Comment by MorrisJobke** [comment]
@brantje Could you add the bot as admin? So it can push to a protected master branch ;)

**Comment by brantje** [comment]
Oh crap, forgot that it was protected. Let me check how i can make the bot an admin.

**Comment by brantje** [comment]
I've unprotected the branch for now, don't see an option to make @nextcloud-bot admin.

**Comment by MorrisJobke** [comment]
> I've unprotected the branch for now, don't see an option to make @nextcloud-bot admin.

Please keep it protected. You need to add the nextcloud bot as collaborator and then set the permission from "write" to "admin".

**Comment by MorrisJobke** [comment]
It was not about the protected-ness, but rather that admins can ignore the other branch enforcements (like one needed review or passed CI checks).

**Comment by brantje** [comment]
Thats the thing, i don't see that option:
image

**Comment by MorrisJobke** [comment]
> Thats the thing, i don't see that option:

-.- This is sadly not possible for personal repositories. Then at least protect the branch again but remove all the needed checks:

bildschirmfoto 2018-04-12 um 10 39 19

**Comment by brantje** [comment]
Done! Github should enable that option for personal repo's imo.

**Comment by MorrisJobke** [comment]
https://github.com/brantje/nextnote/commit/e6dddb05a46dd39e0e722a62fbf31031698c9b49

**Comment by brantje** [comment]
Awesome thanks! How ever i'm seeing strings like `Quick Save` which are in l10n but not my app.
Not that it matter, just something i noticed.

**Comment by brantje** [comment]
Oh didn't know i could just remove them. Thanks again =)

**Comment by MorrisJobke** [comment]
it will clean up automatically all unused stuff. ;) So it's the automatic approach and you should not care anymore about anything from now on ;)

**Comment by brantje** [comment]
Ah nice.

**Comment by brantje** [comment]
@MorrisJobke Is there somehere an overview of what langauge codes `OC.getLocale()` could return? Or even better would it be possible to get the locale like this: `en_GB` or `en_CA`?

**Comment by MorrisJobke** [comment]
> @MorrisJobke Is there somehere an overview of what langauge codes OC.getLocale() could return? Or even better would it be possible to get the locale like this: en_GB or en_CA?

Depends on which languages are translated more than 50% (for the `core` resource), but both you named don't have a team yet so they will not be returned as of now, but it could be in the future. As of now this is the list: https://github.com/nextcloud/server/tree/master/core/l10n

But be aware that it could change anytime depending on the translation status.

**Comment by brantje** [comment]
Good point. 
I'll describe my problem.
I use tinymce for the editor, so logically i need to change that language too, which is done by the `language` parameter, which fetches an external JS file.

Here is an overview of the language folder: 
image
As you see, Nextcloud and TinyMCE use different 'keys' to identify them.

I think a mapper would do, but if i could get away with something like `OC.getLocale().str_replace('-','_')` that would be better.

**Comment by MorrisJobke** [comment]
For our moment we just try to set it and in case it is there is uses it - in case there is no such language it just uses English. So there is no mapping needed because the language code are the same (also for tinymce they are the same). So should work there too.

**Pull request merged**